### PR TITLE
Update mocha from 8.4 to 10.0

### PR DESCRIPTION
**Pull request by CraigMacomber**
# Update mocha from 8.4 to 10.0
## Description

Update mocha from 8.4 to 10.0

## PR Checklist

> Use the check-list below to ensure your branch is ready for PR. If the item is not applicable, leave it blank.

-   [ ] I have updated the documentation accordingly.
-   [ ] I have added tests to cover my changes.
-   [ ] All new and existing tests passed.
-   [ ] My code follows the code style of this project.
-   [ ] I ran the lint checks which produced no new errors nor warnings for my changes.
-   [ ] I have checked to ensure there aren't other open Pull Requests for the same update/change.

## Does this introduce a breaking change?

-   [ ] Yes
-   [x] No

There is a slight risk that updating to a version of @fluid-tools/benchmark after this change will cause issues: if this is the case, the user should update to mocha 10 as well. The breakages in mocha 9 and 10 seem to just be removal of features we don't use, and does not change APIs we do use, so this seems unlikely to be an issue.

The one known out of tree user of @fluid-tools/benchmark (Whiteboard) specifically requested this change to help migrate off of older mocha: it is unlikely this change will be a problem for them.

## Testing

CI testing covers most of what matters here, but benchmark is a special case: benchmark takes a non dev dependency on mocha, and exports a mocha text reporter.

The old reporter works fine with new mocha (tested running experimental/tree's perf tests which use the benchmark package and can be run using the commands shown in #10309). I suspect the reverse is true (making this non breaking) but has not been confirmed.

The new version of the reporter was tested by running `npm run perf` in tools/benchmark/package.json.

## Any relevant logs or outputs

> -   Use this section to provide either screenshots or output of logs as code snippets

## Other information or known dependencies

After this PR is complete, a new version of @fluid-tools/benchmark should be published.
Then the client mono-repo (currently just experimental/tree) should be updated to use it.

### CODE REVIEW

File: tools/benchmark/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@fluid-tools/benchmark",
-  "version": "0.40.0",
+  "version": "0.41.0",
```
Comment (CraigMacomber) [inline tools/benchmark/package.json:3]: Here is the version bump for publishing the new fluid-tools/benchmark: everything else in this update is just updating the versions of mocha and @types/moch used.

File: common/lib/container-definitions/package.json [hunk 2]
```diff
@@ -52,7 +52,7 @@
     "@fluidframework/eslint-config-fluid": "^0.28.2000-0",
     "@microsoft/api-extractor": "^7.22.2",
     "@rushstack/eslint-config": "^2.5.1",
-    "@types/mocha": "^8.2.2",
+    "@types/mocha": "^9.1.1",
```
Comment (DLehenbauer) [inline common/lib/container-definitions/package.json:55]: Curious that this package has a dependency on @types/mocha, but not mocha itself.  Try removing?

File: experimental/PropertyDDS/packages/property-proxy/package.json [hunk 3]
```diff
@@ -46,7 +46,7 @@
     "@fluidframework/build-common": "^0.23.0",
     "@fluidframework/eslint-config-fluid": "^0.28.2000-0",
     "@rushstack/eslint-config": "^2.5.1",
-    "@types/mocha": "^8.2.2",
+    "@types/mocha": "^9.1.1",
```
Comment (DLehenbauer) [inline experimental/PropertyDDS/packages/property-proxy/package.json:49]: ditto for the next few packages.

File: experimental/PropertyDDS/packages/property-query/package.json [hunk 4]
```diff
@@ -39,7 +39,7 @@
     "@fluidframework/mocha-test-setup": "^0.59.4000",
     "chai": "^4.2.0",
     "cross-env": "^7.0.2",
-    "mocha": "^8.4.0",
+    "mocha": "^10.0.0",
```
Comment (DLehenbauer) [inline experimental/PropertyDDS/packages/property-query/package.json:42]: Missing @types dependency?

File: experimental/PropertyDDS/services/property-query-service/package.json [hunk 5]
```diff
@@ -88,7 +88,7 @@
     "eslint-plugin-node": "6.0.1",
     "get-port": "4.2.0",
     "jsdoc": "3.6.7",
-    "mocha": "^8.4.0",
+    "mocha": "^10.0.0",
```
Comment (DLehenbauer) [inline experimental/PropertyDDS/services/property-query-service/package.json:91]: ditto.

File: experimental/dds/tree/package.json [hunk 6]
```diff
@@ -84,7 +84,7 @@
     "eslint-plugin-react": "~7.28.0",
     "eslint-plugin-tsdoc": "~0.2.14",
     "eslint-plugin-unicorn": "~40.0.0",
-    "mocha": "^8.4.0",
+    "mocha": "^10.0.0",
```
Comment (DLehenbauer) [inline experimental/dds/tree/package.json:87]: Missing `@types`?

**Review by DLehenbauer — APPROVED** [review]
Pre-approved.

The diff points out some missing/extraneous `@types/mocha` dependencies that would be convenient to opportunistically clean up.

File: common/lib/container-definitions/package.json (hunk 2 above)
Comment (CraigMacomber) [reply common/lib/container-definitions/package.json]: It does appear that mocha is not used in this directory at all. Removed.

File: experimental/PropertyDDS/packages/property-query/package.json (hunk 4 above)
Comment (CraigMacomber) [reply experimental/PropertyDDS/packages/property-query/package.json:42]: yep. Adding it fixes intelisense.

File: experimental/PropertyDDS/packages/property-proxy/package.json (hunk 3 above)
Comment (CraigMacomber) [reply experimental/PropertyDDS/packages/property-proxy/package.json]: removed

**Pull request merged**